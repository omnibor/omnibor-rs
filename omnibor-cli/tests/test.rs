use insta::Settings;
use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
use std::process::Command;

macro_rules! settings {
    ($block:expr) => {
        let mut settings = Settings::clone_current();
        settings.add_filter(r#"omnibor(?:\.exe)?"#, "omnibor");
        settings.bind(|| $block);
    };
}

#[test]
fn no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")));
    });
}

#[test]
fn artifact_no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).arg("id"));
    });
}

#[test]
fn manifest_no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).arg("manifest"));
    });
}

#[test]
fn debug_no_args() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).arg("debug"));
    });
}

#[test]
fn artifact_id_plain() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).args([
            "id",
            "create",
            "--format",
            "plain",
            "--path",
            "tests/data/main.c"
        ]))
    });
}

#[test]
fn artifact_id_short() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).args([
            "id",
            "create",
            "--format",
            "short",
            "--path",
            "tests/data/main.c"
        ]))
    });
}

#[test]
fn artifact_id_json() {
    settings!({
        assert_cmd_snapshot!(Command::new(get_cargo_bin("omnibor")).args([
            "id",
            "create",
            "--format",
            "json",
            "--path",
            "tests/data/main.c"
        ]))
    });
}
