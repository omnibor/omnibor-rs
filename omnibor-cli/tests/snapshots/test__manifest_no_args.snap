---
source: omnibor-cli/tests/test.rs
info:
  program: omnibor
  args:
    - manifest
---
success: false
exit_code: 2
----- stdout -----

----- stderr -----
Create Input Manifests

Usage: omnibor manifest [OPTIONS] <COMMAND>

Commands:
  create  Create a new manifest and add it to the store
  help    Print this message or the help of the given subcommand(s)

Options:
  -f, --format <FORMAT>  Output format [env: OMNIBOR_FORMAT=] [possible values: plain, short, json]
  -d, --dir <DIR>        Directory to store manifests [env: OMNIBOR_DIR=]
  -c, --config <CONFIG>  Path to a configuration file [env: OMNIBOR_CONFIG=]
  -D, --debug-console    Turn on 'tokio-console' debug integration [env: OMNIBOR_DEBUG_CONSOLE=]
  -v, --verbose...       Increase logging verbosity
  -q, --quiet...         Decrease logging verbosity
  -h, --help             Print help (see more with '--help')
  -V, --version          Print version
